### PR TITLE
[chore] add deploy action to workspace api

### PR DESCRIPTION
**Pull request by lgarciaaco**
Add deploy action. Manage camelk resources

### CODE REVIEW

File: internal/controller/sco/workspace_controller_action_deploy.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+					APIVersion: rr.Resource.APIVersion,
+					Kind:       rr.Resource.Kind,
+					Name:       rr.Resource.Name,
+				},
+			},
+			Labels: map[string]string{
+				"app.kubernetes.io/instance":   rr.Resource.Name,
+				"app.kubernetes.io/part-of":    "sco-operator",
+				"app.kubernetes.io/managed-by": "sco-operator",
+				"app.kubernetes.io/created-by": "sco-operator",
```
Comment (lburgazzoli) [inline internal/controller/sco/workspace_controller_action_deploy.go]: nitpick: you can use constants defined here https://github.com/sco1237896/sco-operator/blob/main/pkg/controller/types.go#L27-L31

File: internal/controller/sco/workspace_controller_action_deploy.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+			},
+		},
+	}
+
+	or, err := controllerutil.CreateOrUpdate(ctx, rr.Client, &ip, func() error {
```
Comment (lburgazzoli) [inline internal/controller/sco/workspace_controller_action_deploy.go:90]: can you open an issue to migrate so server side apply when issues with dependencies are sorted out ?

**Review by lburgazzoli — DISMISSED** [review]
Added some minor comments

File: internal/controller/sco/workspace_controller_action_deploy.go (hunk 2 above)
Comment (lgarciaaco) [reply internal/controller/sco/workspace_controller_action_deploy.go:90]: I will take care

File: pkg/controller/types.go [hunk 3]
```diff
@@ -29,6 +29,7 @@ const (
 	KubernetesLabelAppComponent = "app.kubernetes.io/component"
 	KubernetesLabelAppPartOf    = "app.kubernetes.io/part-of"
 	KubernetesLabelAppManagedBy = "app.kubernetes.io/managed-by"
+	KubernetesLabelCreatedBy    = "app.kubernetes.io/created-by"
```
Comment (lburgazzoli) [inline pkg/controller/types.go]: nitpick: this label is deprecated https://kubernetes.io/docs/reference/labels-annotations-taints/#app-kubernetes-io-created-by-deprecated

**Review by lburgazzoli — APPROVED** [review]

**Pull request merged**